### PR TITLE
openjdk22: add version test

### PR DESCRIPTION
**Pull request by jlesquembre**
## Description of changes
Add version test

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/openjdk/22.nix [hunk 1]
```diff
@@ -254,6 +255,14 @@ stdenv.mkDerivation (finalAttrs: {
   pos = __curPos;
   meta = import ./meta.nix lib featureVersion;
 
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/java";
+  versionCheckProgramArg = "${./JavaGetVersionTest.java}";
```
Comment (doronbehar) [inline pkgs/development/compilers/openjdk/22.nix:262]: Why isn't `java --version` not good?

Comment (jlesquembre) [reply pkgs/development/compilers/openjdk/22.nix]: right, `java --version` is good enough, I'm removing the `JavaGetVersionTest.java` file.

**Comment by doronbehar** [comment]
@ofborg build openjdk22

**Review by doronbehar — APPROVED** [review]
Should be good to go after CI is green. Just out of curiosity, why did you felt this change is needed?

**Comment by jlesquembre** [comment]
> why did you felt this change is needed?

No particular reason, it was originally part of #306616, and I still believe that having a minimal test to ensure that the generated binary can be executed is valuable.

**Comment by doronbehar** [comment]
Hmm OK, would you like to add this test maybe to the other `openjdk{17,21}` etc attributes?

**Comment by jlesquembre** [comment]
> would you like to add this test maybe to the other openjdk{17,21} etc attributes?

I want, but my plan is to extract first the duplicated parts in `openjdk{17..22}` to a common function.

**Comment by doronbehar** [comment]
> > would you like to add this test maybe to the other openjdk{17,21} etc attributes?
> 
> I want, but my plan is to extract first the duplicated parts in `openjdk{17..22}` to a common function.

Hmm OK, so I'll merge this in the meantime. I'm not totally sure that this deduplication is worth it, but feel free to ask me for a review anyway.

**Pull request merged**